siphon/catalog.py

```diff
@@ -276,9 +276,10 @@ def __init__(self, catalog_url):
 
         """
         self.session = session_manager.create_session()
+        session_manager.set_session_options(**kwargs)
```

Comment:
This line is needed for my proxy setup.  I use this line to modify the HTTP headers on the request session.  If you look inside `http_util.py`, the `set_session_options` function adds the keywords I need.  Specifically the "User-Agent" parameter in the header.

Reply:
Interesting... if I comment out `set_session_options` **after** `create_session()` my code fails from the proxy blocking the request.  I expect this.

If I move the `set_session_options` line **before** `create_session()` my code fails from the proxy blocking the request on Line 282 `self.session.get(catalog_url, **kwargs)`.  The same failure if comment out `set_session_options`.  This is unexpected based on what you told me.

I'm not going to pretend like I understand what is going on.  I just know the code in this PR is the only thing that works for my network.

Reply:
Can you show the relevant sample of code you're using with siphon? At least the code that creates a `TDSCatalog` up to the point that fails? I feel like there's something I'm missing here... (You can add that in a stand-alone comment below since I think we're getting a little long-winded here)

Reply:
Just added another comment with some relevant code samples.  Sorry it took so long, ran into some issues with `cftime` package updating and manifesting another bug in `pvlib`.